### PR TITLE
feat: added native fee in structures

### LINKED ISSUE #47: [OmniBridge]: Add nativeFee into structures in the relayer

### PR DESCRIPTION
**Pull request by frolvanya**
* https://github.com/Near-One/omni-bridge/issues/47
* Add chain_id for wormhole
* Rename eth -> evm

### CODE REVIEW

File: omni-relayer/src/workers/near.rs [hunk 1]
```diff
@@ -56,7 +137,7 @@ pub async fn sign_transfer(
                             .sign_transfer(
                                 transfer_message.origin_nonce.into(),
                                 Some(config.near.token_locker_id),
-                                transfer_message.fee.into(),
+                                transfer_message.fee.fee.into(),
```
Comment (olga24912) [inline omni-relayer/src/workers/near.rs:140]: We should provide the full fee here. We might need to fix the rainbow-bridge-sdk first to do this.

https://github.com/Near-One/omni-bridge/blob/main/near/nep141-locker/src/lib.rs#L365

**Review by olga24912 — CHANGES_REQUESTED** [review]

File: omni-relayer/src/workers/near.rs [hunk 2]
```diff
@@ -4,12 +4,93 @@ use anyhow::Result;
...
+                            .await;
+                        }
+                    }
+                }));
+            }
+        }
+
+        join_all(handlers).await;
+
+        tokio::time::sleep(tokio::time::Duration::from_secs(3600)).await;
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs]: This logic was completely removed in f12a3a0

File: omni-relayer/src/workers/near.rs [hunk 3]
```diff
@@ -4,12 +4,93 @@ use anyhow::Result;
...
+                        ) {
+                            info!("Fee is now sufficient");
+
+                            utils::redis::remove_event(
+                                &mut redis_connection,
+                                utils::redis::NEAR_BAD_FEE_EVENTS,
+                                &key,
+                            )
+                            .await;
+                            utils::redis::add_event(
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs]: This logic was completely removed in f12a3a0

File: omni-relayer/src/workers/near.rs [hunk 4]
```diff
@@ -4,12 +4,93 @@ use anyhow::Result;
...
 use log::{error, info, warn};
 
+use near_jsonrpc_client::JsonRpcClient;
 use near_primitives::borsh;
 use omni_connector::OmniConnector;
 use omni_types::{locker_args::ClaimFeeArgs, near_events::Nep141LockerEvent};
 
 use crate::{config, utils};
 
+pub async fn check_bad_fees(
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs]: Fixed in f12a3a0

File: omni-relayer/src/workers/near.rs [hunk 5]
```diff
@@ -4,12 +4,93 @@ use anyhow::Result;
...
+    redis_client: redis::Client,
+    jsonrpc_client: JsonRpcClient,
+) -> Result<()> {
+    let redis_connection = redis_client.get_multiplexed_tokio_connection().await?;
+
+    loop {
+        let mut redis_connection_clone = redis_connection.clone();
+        let Some(events) = utils::redis::get_events(
+            &mut redis_connection_clone,
+            utils::redis::NEAR_BAD_FEE_EVENTS.to_string(),
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs]: Fixed in 3abf415

File: omni-relayer/src/workers/near.rs (hunk 1 above)
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs:59]: Will be fixed after merging https://github.com/Near-One/rainbow-bridge-sdk-rs/pull/39

File: omni-relayer/src/utils/near.rs [hunk 6]
```diff
@@ -46,39 +46,51 @@ pub async fn handle_streamer_message(
...
+                                transfer_message.origin_nonce.0.to_string(),
+                                log,
+                            )
+                            .await;
+                        } else {
+                            warn!("Fee is not sufficient for transfer: {:?}", transfer_message);
+
+                            utils::redis::add_event(
+                                redis_connection,
+                                utils::redis::NEAR_BAD_FEES,
```
[the remark this answers is not in the document]
Comment (frolvanya) [reply omni-relayer/src/utils/near.rs]: Fixed in 1316b69

File: omni-relayer/src/workers/near.rs (hunk 1 above)
Comment (frolvanya) [reply omni-relayer/src/workers/near.rs]: Fixed in 77a35c0 (still changes aren't merged in bridge-sdk, but it works locally)

File: omni-relayer/src/utils/near.rs [hunk 7]
```diff
@@ -46,39 +46,35 @@ pub async fn handle_streamer_message(
...
             } => {
-                // TODO: If fee is insufficient, it should be handled later. For example,
-                // add to redis and try again in 1 hour
                 match utils::fee::is_fee_sufficient(
                     jsonrpc_client,
                     &transfer_message.sender,
                     &transfer_message.recipient,
                     &transfer_message.token,
-                    transfer_message.fee.into(),
+                    transfer_message.fee.fee.into(),
```
Comment (olga24912) [inline omni-relayer/src/utils/near.rs:57]: Actually, I think it would be a good idea to pass the native fee here as well.

**Review by olga24912 — APPROVED** [review]

File: omni-relayer/src/utils/near.rs (hunk 7 above)
Comment (frolvanya) [reply omni-relayer/src/utils/near.rs:57]: I want to make all changes related to fee calculation in the next PR. This whole function must be rewritten from scratch as well as its signature

**Pull request merged**